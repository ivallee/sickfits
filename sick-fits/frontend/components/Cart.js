import React from 'react';
import { Query, Mutation } from 'react-apollo';
import gql from 'graphql-tag';
import User from './User';
import CartStyles from './styles/CartStyles';
import Supreme from './styles/Supreme';
import CloseButton from './styles/CloseButton';
import SickButton from './styles/SickButton';

const LOCAL_STATE_QUERY = gql`
  query {
    cartOpen @client
  }
`;

const TOGGLE_CART_MUTATION = gql`
  mutation {
    toggleCart @client
  }
`;


const Cart = () => {
  return (
    <User>
      {({data: { me }}) => {
        console.log(me);
        if (!me) return null;
        return (
          <Mutation mutation={TOGGLE_CART_MUTATION}>
            {toggleCart =>  (
              <Query query={LOCAL_STATE_QUERY}>
                {({data}) => (
                  <CartStyles open={data.cartOpen}>
                    <header>
                      <CloseButton onClick={toggleCart} title="close">&times;</CloseButton>
                      <Supreme>{me.name}'s Cart</Supreme>
                      <p>You have {me.cart.length} item{me.cart.length === 1 ? '' : 's'} in your cart</p>
                    </header>
                    <ul>
                      {me.cart.map(cartItem => {
                       return <li>{cartItem.id}</li>
                      })}
                    </ul>
                    <footer>
                      <p>19.99</p>
                      <SickButton>Checkout</SickButton>
                    </footer>
                  </CartStyles>
                )}
              </Query>
            )}
          </Mutation>
        );
      }}
    </User>
  );
};

export default Cart;
export { LOCAL_STATE_QUERY, TOGGLE_CART_MUTATION };