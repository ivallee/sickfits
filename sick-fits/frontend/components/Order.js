import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Query } from 'react-apollo';
import { format } from 'date-fns';
import Head from 'next/head';
import gql from 'graphql-tag';
import formatMoney from '../lib/formatMoney';
import OrderStyles from './styles/OrderStyles';
import Error from './ErrorMessage';
import OrderItem from './OrderItem';

const SINGLE_ORDER_QUERY = gql`
  query SINGLE_ORDER_QUERY($id: ID!){
    order(id: $id){
      id
      charge
      total
      createdAt
      user{
        id
      }
      items {
        id
        title
        description
        price
        image
        quantity
      }
    }
  }
`;

class Order extends Component {
  static propTypes = {
    id: PropTypes.string.isRequired
  };
  render() {
    return (
      <Query query={SINGLE_ORDER_QUERY} variables={{ id: this.props.id }}>
      {({data, error, loading}) => {
        if (error) return <Error error={error} />
        if (loading) return <p>Loading...</p>
        const order = data.order;
        return (
          <OrderStyles data-test="order">
            <Head>
              <title>Sit Fits - Order {order.id}</title>
            </Head>
            <p>
              <span>Order ID:</span> 
              <span>{this.props.id}</span>
            </p>
            <p>
              <span>Date</span>
              <span>{format(order.createdAt, 'MMMM d, YYYY h:mm a')}</span>
            </p>
            <p>
              <span>Order Total</span>
              <span>{formatMoney(order.total)}</span>
            </p>
            <p>
              <span>Item Count</span>
              <span>{order.items.length}</span>
            </p>
            <div>
              {order.items.map(item => (
                <OrderItem key={item.id} item={item}/>
              ))}
            </div>
          </OrderStyles>
        );
      }}
      </Query>
    );
  }
}

export default Order;
export { SINGLE_ORDER_QUERY };