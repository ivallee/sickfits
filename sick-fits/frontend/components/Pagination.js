import React from 'react';
import PaginationStyles from './styles/PaginationStyles';

const Pagination = props => {
  return (
    <PaginationStyles>
      <p>I am the pagination</p>
    </PaginationStyles>
  );
};

export default Pagination;