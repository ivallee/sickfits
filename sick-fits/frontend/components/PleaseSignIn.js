import { Query } from 'react-apollo';
import { CURRENT_USER_QUERY } from './User';
import Signin from './Signin';

const PleaseSignIn = props => (
  <Query query={CURRENT_USER_QUERY}>
    {({data, loading}) => {
      if (loading) return <p>Loading...</p>
      if (!data.me) {
        return <Signin />
      }
      return <div>{props.children}</div>
    }}
  </Query>
);

export default PleaseSignIn;
