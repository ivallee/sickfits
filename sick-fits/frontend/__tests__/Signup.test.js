import { mount } from 'enzyme';
import wait from 'waait';
import toJSON from 'enzyme-to-json';
import { MockedProvider } from 'react-apollo/test-utils';
import { ApolloConsumer } from 'react-apollo';
import Signup, { SIGNUP_MUTATION } from '../components/Signup';
import { CURRENT_USER_QUERY } from '../components/User';
import { fakeUser } from '../lib/testUtils';

function type(wrapper, name, value) {
  wrapper.find(`input[name="${name}"]`).simulate('change', {
    target: { name, value },
  });
}

const me = fakeUser();
const mocks = [
  // Signup mutation mock
  {
    request: {
      query: SIGNUP_MUTATION,
      variables: {
        name: me.name,
        email: me.email,
        password: 'testpassword',
      },
    },
    result: {
      data: {
        signup: {
          __typename: 'User',
          id: 'abc123',
          email: me.email,
          name: me.name,
        },
      },
    },
  },
  // current user query mock
  {
    request: { query: CURRENT_USER_QUERY },
    result: { data: { me } },
  }
];

describe('<Signup/>', () => {
  it('renders and mathces snapshot', () => {
    const wrapper = mount(
      <MockedProvider>
        <Signup />
      </MockedProvider>
    );
    const form = wrapper.find('form[data-test="form"]');
    expect(toJSON(form)).toMatchSnapshot();
  });

  it('calls the signup mutation properly', async () => {
    let apolloClient;
    const wrapper = mount(
      <MockedProvider mocks={mocks}>
        <ApolloConsumer>
          {client => {
            apolloClient = client;
            return <Signup/>;
          }}
        </ApolloConsumer>
      </MockedProvider>
    );
    await wait();
    wrapper.update();
    type(wrapper, 'name', me.name);
    type(wrapper, 'email', me.email);
    type(wrapper, 'password', 'testpassword');
    wrapper.update();
    wrapper.find('form').simulate('submit');
    await wait();
    wrapper.update();
    const user = await apolloClient.query({ query: CURRENT_USER_QUERY });
    expect(user.data.me).toMatchObject(me);
  });
});