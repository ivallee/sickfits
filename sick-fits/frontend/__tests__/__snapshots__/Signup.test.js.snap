// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Signup/> renders and mathces snapshot 1`] = `
<form
  className="Form-sc-1xszr8q-0 krRxky"
  data-test="form"
  method="post"
  onSubmit={[Function]}
>
  <fieldset
    aria-busy={false}
    disabled={false}
  >
    <h2>
      Sign Up For an Account
    </h2>
    <DisplayError
      error={Object {}}
    />
    <label
      htmlFor="name"
    >
      name
      <input
        name="name"
        onChange={[Function]}
        placeholder="name"
        type="text"
        value=""
      />
    </label>
    <label
      htmlFor="email"
    >
      email
      <input
        name="email"
        onChange={[Function]}
        placeholder="email"
        type="email"
        value=""
      />
    </label>
    <label
      htmlFor="password"
    >
      password
      <input
        name="password"
        onChange={[Function]}
        placeholder="password"
        type="password"
        value=""
      />
    </label>
    <button
      type="submit"
    >
      Sign up!
    </button>
  </fieldset>
</form>
`;
